lib/salus/report.rb

```diff
@@ -228,17 +228,13 @@ def satisfies_filter?(directive)
       return true if @report_filter == 'all'
       return false if @report_filter == 'none'
 
-      # Use regex to capture two parts of the filter, delimited by a ':'
-      filter_regex = Regexp.new('([a-zA-Z0-9]+):?([a-zA-Z0-9\*]*)')
-      captured_parameters = filter_regex.match(@report_filter)
-      directive_key = captured_parameters[1]
-      directive_value = directive[directive_key]
-      # Since the value of the second capture group can follow a regex
-      # pattern (i.e. can use wildcards), we build a regex from it
-      # and use it to match against the directive value
-      directive_value_regex = Regexp.new(captured_parameters[2])
-
-      directive.key?(directive_key) && directive_value_regex.match?(directive_value)
+      recovered_values = @report_filter.split(':')
```

Comment:
This `split` probably works for now (assuming there is only one `:`).  But if there is more than one `:` (maybe in the future we'll have something that has more than one `:`), then the code will break.

So it would be safer to use the code I mentioned earlier
```ruby
first_colon = index of the first colon
key = @report_filter[0... first_colon - 1]
value = @report_filter[first_colon + 1 ... end]
```

Reply:
Actually, I think it's already possible to have more than one `:` now, like
```ruby
uri:http://
```

Reply:
Used the second argument of `split?()` to achieve this.
`'a:b:c:d://:e:'.split(':', 2)` returns `["a", "b:c:d://:e:"]` for example.